# Problem 1:

In each of the following situations, state whether it is a correctly stated hypothesis
testing problem and why?
1. 𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25
2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10
3. 𝐻0: 𝑥 = 50, 𝐻1: 𝑥 ≠ 50
4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5
5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30

##### 1. 𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25

The above expression is a correctly stated hypothesis for testing problem. Because both the statements are mutually exclusive and covers all the possible scenerios.

##### 2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10

The above expression is NOT correctly stated hypothesis for testing problem. Because both the statements are NOT mutually exclusive and  does not cover all the possible scenerios(i.e what to do in case 𝜎 < 10).

##### 3. 𝐻0: 𝑥 = 50, 𝐻1: 𝑥 ≠ 50

The above expression is a correctly stated hypothesis for testing problem. Because both the statements are mutually exclusive and covers all the possible scenerios.

##### 4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5

The above expression is NOT correctly stated hypothesis for testing problem. Because both the statements are NOT mutually exclusive and  does not cover all the possible scenerios(i.e what to do if p comes out to be a value other than 0.1 or 0.5).

##### 5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30

The above expression is NOT correctly stated hypothesis for testing problem. Because both the statements are NOT mutually exclusive and  does not cover all the possible scenerios.(i.e what to do in case s < 30).

# Problem 2:

The college bookstore tells prospective students that the average cost of its
textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics
students thinks that the average cost is higher. To test the bookstore’s claim against
their alternative, the students will select a random sample of size 100. Assume that
the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the
5% level of significance and state your decision.

In [1]:
from scipy.stats import norm
import math
import numpy as np
import scipy.stats

##### Hypothesis
H0 : mean(𝜇) <= Rs. 52  
H1 : mean(𝜇) > Rs. 52

In [2]:
mean = 52
std_dev = 4.50

sample_mean = 52.80
# Sample_std_dev = 𝜎/math.sqrt(sample size)
sample_std_dev = std_dev/(math.sqrt(100))

z = (sample_mean - mean)/sample_std_dev

# Level of significance is 5%. Hence p = 0.05
p = 0.05
area_on_left = 1-p
z_critical = norm.ppf(area_on_left)

print(f"Calculated value of z from sample is {z}")
print(f"Calculated value of z should be less than {z_critical} in order to accept null hypothesis")

Calculated value of z from sample is 1.7777777777777715
Calculated value of z should be less than 1.6448536269514722 in order to accept null hypothesis


##### Conclusion
Since calculated z value (i.e 1.77) is more than critical z value (i.e 1.64).  
Hence the null hypothesis is rejected.  
There is enough evidence at the 5% level of significance to suggest that the average cost of textbooks is greater than Rs. 52 . 

# Problem 3:

A certain chemical pollutant in the Genesee River has been constant for several
years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A
group of factory representatives whose companies discharge liquids into the river is
now claiming that they have lowered the average with improved filtration devices. A
group of environmentalists will test to see if this is true at the 1% level of
significance. Assume \ that their sample of size 50 gives a mean of 32.5 ppm.
Perform a hypothesis test at the 1% level of significance and state your decision.

##### Hypothesis
H0 : mean(𝜇) >= 34 ppm  
H1 : mean(𝜇) < 34 ppm

In [3]:
mean = 34
std_dev = 8

sample_mean = 32.5
sample_size = 50
sample_std_dev = std_dev/math.sqrt(sample_size)

# Calculation of z-value
z = (sample_mean-mean)/sample_std_dev

# From the problem statement it is a one tail testing on left side of Normal distribution. Level of significance 1%.
p = 0.01

# Calculation of z_critical.
z_critical = norm.ppf(p)

print(f"Calculated z value is {z}")
print(f"Calculated value of z should be greater than {z_critical} in order to accept null hypothesis")

Calculated z value is -1.3258252147247767
Calculated value of z should be greater than -2.3263478740408408 in order to accept null hypothesis


##### Conclusion
Since calculated z value (i.e -1.3258) is more than critical z value (i.e -2.3263).  
Hence the null hypothesis is accepted.  
There is NOT enough evidence at 1% level of significance to suggest that the mean of chemical pollutant has gone down by using improved filter devices as claimed by group of factory representatives. 

# Problem 4:

Based on population figures and other general information on the U.S. population,
suppose it has been estimated that, on average, a family of four in the U.S. spends
about $1135 annually on dental expenditures. Suppose further that a regional dental
association wants to test to determine if this figure is accurate for their area of
country. To test this, 22 families of 4 are randomly selected from the population in
that area of the country and a log is kept of the family’s dental expenditure for one
year. The resulting data are given below. Assuming, that dental expenditure is
normally distributed in the population, use the data and an alpha of 0.05 to test the
dental association’s hypothesis.  
1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994

##### Hypothesis
H0 : mean(𝜇) = \\$1135  
H1 : mean(𝜇) != \\$1135

In [4]:
mean = 1135

# Since the standard deviation is not mentioned for the population data we will use the t-test here for hypothesis testing.
dental_expenditure = [1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994]

sample_mean = np.mean(dental_expenditure)
std_dev = np.std(dental_expenditure)
sample_size = len(dental_expenditure)

sample_std_dev = std_dev/math.sqrt(sample_size)

# Calculated t-value
t = (sample_mean - mean)/sample_std_dev

# Degrees of freedom
df = sample_size - 1

# Calculation of critical t-value for 0.05 level of significance and degree of freedom.
p = 0.05
t_critical = scipy.stats.t.ppf(q = 1-p/2, df = df)

print(f"The calculated t-value is {t}")
print(f"The calculated t-value should lie between {-t_critical} and {t_critical} in order to accept null hypothesis.")

The calculated t-value is -2.070747228595759
The calculated t-value should lie between -2.079613844727662 and 2.079613844727662 in order to accept null hypothesis.


##### Conclusion
Since calculated t-value (i.e -2.07074) lie between -2.0796 and 2.0796.  
Hence the null hypothesis is accepted.  
There is NOT enough evidence at 5% level of significance that dental expenditure mean of US population will not fit for the locality taken into consideration. 

# Problem 5:

In a report prepared by the Economic Research Department of a major bank the
Department manager maintains that the average annual family income on Metropolis is \\$48,432. What do you conclude about the validity of the report if a random sample of 400 families shows and average income of \\$48,574 with a standard deviation of 2000?

##### Hypothesis
H0 : mean(𝜇) = \\$48432  
H1 : mean(𝜇) != \\$48432

In [5]:
mean = 48432
std_dev = 2000

sample_size = 400
sample_mean = 48574
sample_std_dev = std_dev/math.sqrt(sample_size)

#Calculating the z-value
z = (sample_mean - mean)/sample_std_dev

#Calculating the zcritical for 5% level of significance in a two-tail testing
p = 0.05
z_critical = norm.ppf(1-0.05/2)

print(f"The calculated z-value is {z}.")
print(f"The expected value of z should be between {-z_critical} and {z_critical} in order to accept the null hypothesis.")

The calculated z-value is 1.42.
The expected value of z should be between -1.959963984540054 and 1.959963984540054 in order to accept the null hypothesis.


##### Conclusion
Since calculated z-value (i.e 1.42) lie between -1.9599 and 1.9599.  
Hence the null hypothesis is accepted.  
There is NOT enough evidence at 5% level of significance that the average annual family income on Metropolis deviates from \\$48,432 . 

# Problem 6:

Suppose that in past years the average price per square foot for warehouses in the
United States has been \\$32.28. A national real estate investor wants to determine
whether that figure has changed now. The investor hires a researcher who randomly
samples 19 warehouses that are for sale across the United States and finds that the
mean price per square foot is \\$31.67, with a standard deviation of \\$1.29. assume
that the prices of warehouse footage are normally distributed in population. If the
researcher uses a 5% level of significance, what statistical conclusion can be
reached? What are the hypotheses?

##### Hypothesis
H0 : mean(𝜇) = \\$32.28   
H1 : mean(𝜇) != \\$32.28

In [6]:
mean = 32.28
std_dev = 1.29

sample_size = 19
sample_mean = 31.67
sample_std_dev = std_dev/math.sqrt(sample_size)

# z-calculated value
z = (sample_mean-mean)/sample_std_dev

# Here the level of significance is 5%.
p=0.05
z_critical = norm.ppf(1-0.05/2)

print(f"The calculated z-value is {z}")
print(f"The calculated z-value should lie between {z_critical} and {-z_critical} in order to accept the null hypothesis.")

The calculated z-value is -2.06118477175179
The calculated z-value should lie between 1.959963984540054 and -1.959963984540054 in order to accept the null hypothesis.


##### Conclusion
Since calculated z-value (i.e -2.0611) does NOT lie between -1.9599 and 1.9599  
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance that the average price per square foot for warehouses in the United States is NOT $32.28.

# Problem 7:

Fill in the blank spaces in the table and draw your conclusions from it.

##### a) Acceptance Region : 48.5 < x̄ < 51.5 , Sample size = 10

###### α 
This Question has insufficient data it's NOT possible to find alpha without the help of standard deviation.    
In order to find alpha value we require:    
a) Acceptance region (Given in Q)  
b) Sample size (Given in Q)  
c) Standard deviation (NOT Given)  

Let me explain how standard deviation will help in finding α :  
    Let's say std_dev = 21  
    mean = (48.5 + 51.5)/2  
         = 50  
    sample_size = 10  
    
    z = (48.5 - mean)/(std_dev/math.sqrt(sample_size))
    z = (48.5- 50)/(21/math.sqrt(10))
    z = -0.2258
    
For finding alpha value we are multiplying by 2 becoz its a two tail test:  
    α = 2 * norm.cdf(z)  
      = 0.8212
###### β at μ = 52


###### β at μ = 50.5

# Problem 8:

Find the t-score for a sample size of 16 taken from a population with mean 10 when
the sample mean is 12 and the sample standard deviation is 1.5.

In [7]:
mean = 10

sample_size = 16
sample_mean = 12
sample_std_dev = 1.5

t = (sample_mean - mean)/(sample_std_dev/math.sqrt(sample_size))

print(f"The t-score of the above data is {t}")

The t-score of the above data is 5.333333333333333


# Problem 9:

Find the t-score below which we can expect 99% of sample means will fall if samples
of size 16 are taken from a normally distributed population.

In [8]:
# Level of significance is 1%
p = 0.01

sample_size = 16

t = scipy.stats.t.ppf(q = 1-p, df= sample_size-1)

print(f"The t-score of the above data is {t}")

The t-score of the above data is 2.602480294995493


# Problem 10:

If a random sample of size 25 drawn from a normal population gives a mean of 60
and a standard deviation of 4, find the range of t-scores where we can expect to find
the middle 95% of all sample means. Compute the probability that (−𝑡0.05 <𝑡<𝑡0.10).

In [9]:
sample_size = 25

mean = 60 
std_dev = 4

sample_std_dev = std_dev/math.sqrt(sample_size)

# Confidence interval is 95%. Hence level of significance is 5%
p = 0.05

t_critical = scipy.stats.t.ppf(q = p/2, df = sample_size-1)

print(f"The value of t-score to find the middle 95% of all sample means is between {t_critical} and {-t_critical}")

The value of t-score to find the middle 95% of all sample means is between -2.063898561628021 and 2.063898561628021


##### P(−𝑡0.05 <𝑡<𝑡0.10)

In [10]:
# Probability of normal distribution curve from left till t = -0.05
a1 = scipy.stats.t.cdf(-0.05, df = sample_size-1)

# Probability of normal distribution curve from left till t = 0.10
a2 = scipy.stats.t.cdf(0.10, df = sample_size-1)

probability = a2 - a1

print(f"Probability that (−𝑡0.05 <𝑡<𝑡0.10) is {probability}")

Probability that (−𝑡0.05 <𝑡<𝑡0.10) is 0.05914441613731247


# Problem 11:

Two-tailed test for difference between two population means
Is there evidence to conclude that the number of people travelling from Bangalore to
Chennai is different from the number of people travelling from Bangalore to Hosur in
a week, given the following:  
Population 1: Bangalore to Chennai n1 = 1200  
x1 = 452  
s1 = 212  
Population 2: Bangalore to Hosur n2 = 800  
x2 = 523  
s2 = 185  

##### Hypothesis
μ1 - mean of population 1  
μ2 - mean of population 2   
H0 : μ1 = μ2  
H1 : μ1 != μ2 

In [11]:
# Bangalore to Chennai

# sample size
n1 = 1200
# sample mean
x1 = 452
# sample std deviation
s1 = 212

# Bangalore to Hosur

# sample size
n2 = 800
# sample mean
x2 = 523
# sample std deviation
s2 = 185

standard_error = math.sqrt(s1**2/n1 + s2**2/n2)

z_calculated = (x2-x1)/standard_error

# Considering 5% level of siginificance
p = 0.05
z_critical = norm.ppf(p/2)

print(f"The calculated z score is {z_calculated}")
print(f"The z value is expected to lie between {z_critical} and {-z_critical} in order to reject the null hypothesis.")

The calculated z score is 7.926428526759299
The z value is expected to lie between -1.9599639845400545 and 1.9599639845400545 in order to reject the null hypothesis.


##### Conclusion
Since calculated z-value (i.e 7.9264) does NOT lie between -1.9599 and 1.9599.  
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance that the two population means are not the same.

# Problem 12:

Is there evidence to conclude that the number of people preferring Duracell battery is
different from the number of people preferring Energizer battery, given the following:  
Population 1: Duracell  
n1 = 100  
x1 = 308  
s1 = 84  
Population 2: Energizer  
n2 = 100  
x2 = 254  
s2 = 67  

##### Hypothesis
μ1 - mean of population preferring Duracell battery  
μ2 - mean of population preferring Energizer battery  
H0 : μ1 = μ2  
H1 : μ1 != μ2 

In [12]:
# Population preferring Duraccell battery

# sample size
n1 = 100
# sample mean
x1 = 308
# sample std deviation
s1 = 84

# Population preferring Energizer battery

# sample size
n2 = 100
# sample mean
x2 = 254
# sample std deviation
s2 = 67

standard_error = math.sqrt(s1**2/n1 + s2**2/n2)

z_calculated = (x2-x1)/standard_error

# Considering 5% level of siginificance
p = 0.05
z_critical = norm.ppf(p/2)

print(f"The calculated z score is {z_calculated}")
print(f"The calculated z value should lie between {z_critical} and {-z_critical} in order to accept the null hypothesis.")

The calculated z score is -5.025702668336442
The calculated z value should lie between -1.9599639845400545 and 1.9599639845400545 in order to accept the null hypothesis.


##### Conclusion
Since calculated z-value (i.e -5.0257) does NOT lie between -1.9599 and 1.9599.  
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance to prove that the two population preferring different battery are not the same.

# Problem 13:

Pooled estimate of the population variance
Does the data provide sufficient evidence to conclude that average percentage
increase in the price of sugar differs when it is sold at two different prices?  
Population 1: Price of sugar = Rs. 27.50 n1 = 14  
x1 = 0.317%  
s1 = 0.12%  
Population 2: Price of sugar = Rs. 20.00 n2 = 9  
x2 = 0.21%  
s2 = 0.11%  

##### Hypothesis
μ1 - mean price of sugar at Place A  
μ2 - mean price of sugar at Place B  
H0 : μ1 = μ2  
H1 : μ1 != μ2 

In [13]:
# Information related to place A

# Sample size
n1 = 14
# Mean increase
x1 = 0.317
# Sample std_dev
s1 = 0.12

# Information related to place B

# Sample size
n2 = 9
# Mean increase
x2 = 0.21
# Sample std_dev
s2 = 0.11

# From the above we can conclude that we need to perform two sample t-test since sample size is less.

#std_dev
numerator = (n1-1)*(s1**2) + (n2-1)*(s2**2)
denominator = n1 + n2 - 2

std_dev = math.sqrt(numerator/denominator)

standard_error = std_dev*(math.sqrt(1/n1 + 1/n2))

t_score = (x1 - x2)/standard_error

# Considering 5% level of significance in this two-tailed test.
p = 0.05
t_critical = scipy.stats.t.ppf(p/2, df = n1+n2-2)

print(f"The calculated t_score from the above information is {t_score}")
print(f"The calculated t_score should lie between {t_critical} and {-t_critical} in order to accept the null hypothesis.")

The calculated t_score from the above information is 2.15355322387416
The calculated t_score should lie between -2.0796138447276626 and 2.0796138447276626 in order to accept the null hypothesis.


##### Conclusion
Since calculated z-value (i.e -2.1535) does NOT lie between -2.0796 and 2.0796.  
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance that average percentage increase in the price of sugar differs when it is sold at two different prices.

# Problem 14:

The manufacturers of compact disk players want to test whether a small price
reduction is enough to increase sales of their product. Is there evidence that the
small price reduction is enough to increase sales of compact disk players?  
Population 1: Before reduction  
n1 = 15  
x1 = Rs. 6598  
s1 = Rs. 844  
Population 2: After reduction  
n2 = 12  
x2 = RS. 6870  
s2 = Rs. 669  

##### Hypothesis
μ1 - mean of sales of population 1  
μ2 - mean of sales of population 2  
H0 : μ2 - μ1 <= 0  
H1 : μ2 - μ1 > 0 

In [14]:
# Information related to sale before reduction

# Sample size
n1 = 15
# Mean increase
x1 = 6598
# Sample std_dev
s1 = 844

# Information related to sale after reduction

# Sample size
n2 = 12
# Mean increase
x2 = 6870
# Sample std_dev
s2 = 669

# From the above we can conclude that we need to perform two sample t-test since sample size is less.

#std_dev
numerator = (n1-1)*(s1**2) + (n2-1)*(s2**2)
denominator = n1 + n2 - 2

std_dev = math.sqrt(numerator/denominator)

standard_error = std_dev*(math.sqrt(1/n1 + 1/n2))

t_score = (x2 - x1)/standard_error

# Considering 5% level of significance in this two-tailed test.
p = 0.05
t_critical = scipy.stats.t.ppf(1-p, df = n1+n2-2)

print(f"The calculated t_score from the above information is {t_score}")
print(f"The calculated t_score should be less than {t_critical} in order to accept the null hypothesis.")

The calculated t_score from the above information is 0.9098300343990459
The calculated t_score should be less than 1.7081407612518986 in order to accept the null hypothesis.


##### Conclusion
Since calculated t-value (i.e 0.9098) is less than critical t_score (i.e 1.70814)  
Hence the null hypothesis is accepted.  
There is NOT enough evidence at 5% level of significance that the sales are affeccted by reduction in price.

# Problem 15:

Comparisons of two population proportions when the hypothesized difference is zero
Carry out a two-tailed test of the equality of banks’ share of the car loan market in
1980 and 1995.  
Population 1: 1980  
n1 = 1000  
x1 = 53  
𝑝 1 = 0.53  
Population 2: 1985  
n2 = 100  
x2 = 43  
𝑝 2= 0.53  

##### Hypothesis
μ1 - sample proportion of population 1   
μ2 - sample proportion of population 2  
H0 : μ2 - μ1 = 0    
H1 : μ2 - μ1 != 0    

In [15]:
# Population 1: 1980  

# sample size
n1 = 1000  
# sample mean
x1 = 53  
# sample standard deviation
𝑝1 = 0.53  

# Population 2: 1985  

# sample size
n2 = 100  
# sample mean
x2 = 43  
# sample standard deviation
𝑝2 = 0.53

combined_proportion = (x1 + x2)/(n1 + n2)

standard_error = math.sqrt(combined_proportion*(1-combined_proportion)*(1/n1 + 1/n2))

z_calculated = (p1 - p2)/standard_error

# Considering 5% level of siginificance
p = 0.05
z_critical = norm.ppf(p/2)

print(f"The calculated z-score is {z_calculated}")
print(f"The calculated z-score should lie between {z_critical} and {-z_critical} in order to accept the null hypothesis.")

The calculated z-score is 0.0
The calculated z-score should lie between -1.9599639845400545 and 1.9599639845400545 in order to accept the null hypothesis.


##### Conclusion
Since calculated t-value (i.e 0.0) is lies between -1.9599 and 1.9599.  
Hence the null hypothesis is accepted.  
There is NOT enough evidence at 5% level of significance that there is no statistically significant difference between bank shares in car loan.

# Problem 16: 

Carry out a one-tailed test to determine whether the population proportion of
traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes
are offered as at least 10% higher than the proportion of such buyers when no
sweepstakes are on.  
Population 1: With sweepstakes  
n1 = 300  
x1 = 120  
𝑝1 = 0.40  
Population 2: No sweepstakes  
n2 = 700  
x2 = 140  
𝑝2= 0.20  

##### Hypothesis
μ1 - sample proportion of population 1   
μ2 - sample proportion of population 2  
H0 : μ2 - μ1 <= 0.10    
H1 : μ2 - μ1 >  0.10

In [16]:
# Population 1:With Sweepstakes

# sample size
n1 = 300
# sample mean
x1 = 120
# sample proportion
p1 = 0.40

# Population 2: Without Sweepstakes

# sample size
n2 = 700
# sample mean
x2 = 140
# sample proportion
p2 = 0.2

# From hypothesis
D = 0.10

standard_error = math.sqrt(p1*(1-p1)/n1 + p2*(1-p2)/n2)

z_score = ((p1 - p2) - D) / standard_error

# Considering 5% level of significance
p = 0.05
z_critical = norm.ppf(1-p)

print(f"The calculated value of z_score is {z_score}")
print(f"The calculated z value should be less than {z_critical} in order to accept the null hypothesis.")

The calculated value of z_score is 3.1180478223116177
The calculated z value should be less than 1.6448536269514722 in order to accept the null hypothesis.


##### Conclusion
Since calculated t-value (i.e 3.1118) is more than the z_critical (i.e 1.6448).  
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance that the proportion of customers buying atleast $2500 of travelers checks is atleast 10% higher when sweepstakes are on.

# Problem 17:

A die is thrown 132 times with the following results: Number turned up: 1, 2, 3, 4, 5, 6
Frequency: 16, 20, 25, 14, 29, 28
Is the die unbiased? Consider the degrees of freedom as 𝑝 − 1.

In [17]:
import numpy as np
import pandas as pd

In [18]:
data = {
    'Number' : [1, 2, 3, 4, 5, 6],
    'Obs_freq(O)' : [16, 20, 25, 14, 29, 28]
}
df = pd.DataFrame(data)
df

Number  Obs_freq(O)
0       1           16
1       2           20
2       3           25
3       4           14
4       5           29
5       6           28

##### Hypothesis
H0 : The die is unbiased.  
H1 : The die is biased.

In [19]:
# Expected frequency for 132 throws
exp = 132 / 6

In [20]:
df['Expec_freq(E)'] = exp
df

Number  Obs_freq(O)  Expec_freq(E)
0       1           16           22.0
1       2           20           22.0
2       3           25           22.0
3       4           14           22.0
4       5           29           22.0
5       6           28           22.0

In [21]:
df['(O - E)^2'] = (df['Obs_freq(O)'] - df['Expec_freq(E)'])**2
df

Number  Obs_freq(O)  Expec_freq(E)  (O - E)^2
0       1           16           22.0       36.0
1       2           20           22.0        4.0
2       3           25           22.0        9.0
3       4           14           22.0       64.0
4       5           29           22.0       49.0
5       6           28           22.0       36.0

In [22]:
df['(O - E)^2/E'] = df['(O - E)^2']/df['Expec_freq(E)']
df

Number  Obs_freq(O)  Expec_freq(E)  (O - E)^2  (O - E)^2/E
0       1           16           22.0       36.0     1.636364
1       2           20           22.0        4.0     0.181818
2       3           25           22.0        9.0     0.409091
3       4           14           22.0       64.0     2.909091
4       5           29           22.0       49.0     2.227273
5       6           28           22.0       36.0     1.636364

In [23]:
chi_calculated = df['(O - E)^2/E'].sum()

#Degree of freedom
Dof = df['Number'].count() - 1

# Expected chi_square value at 5% level of significance one-tail testing
chi_expected = scipy.stats.chi2.ppf(1-0.05, df = Dof)

print(f"The calculated chi square value is {chi_calculated}.")
print(f"The  calculated chi square value should be less than {chi_expected} in order to accept the null hypothesis.")

The calculated chi square value is 9.0.
The  calculated chi square value should be less than 11.070497693516351 in order to accept the null hypothesis.


###### Conclusion
Since calculated t-value (i.e 9.0) is less than the z_critical (i.e 11.0704).  
Hence the null hypothesis is accepted.  
There is NOT enough evidence at 5% level of significance in order to prove the die is biased.

# Problem 18:

In a certain town, there are about one million eligible voters. A simple random
sample of 10,000 eligible voters was chosen to study the relationship between
gender and participation in the last election. The results are summarized in the
following 2X2 (read two by two) contingency table:  
We would want to check whether being a man or a woman (columns) is independent of
having voted in the last election (rows). In other words, is “gender and voting independent”?

In [24]:
data = {
    'Status' : ['Voted', 'Not Voted'],
    'Men' : [2792, 1486],
    'Women' : [3591, 2131]
}
df = pd.DataFrame(data)
df

Status   Men  Women
0      Voted  2792   3591
1  Not Voted  1486   2131

###### Hypothesis
H0 : Sex and voting are independent.  
H1 : Sex and voting are dependent.

In [25]:
df.loc[len(df.index)] = ['Totals', df['Men'].sum(), df['Women'].sum()]
df

Status   Men  Women
0      Voted  2792   3591
1  Not Voted  1486   2131
2     Totals  4278   5722

In [26]:
df['Totals'] = df['Men'] + df['Women']
df

Status   Men  Women  Totals
0      Voted  2792   3591    6383
1  Not Voted  1486   2131    3617
2     Totals  4278   5722   10000

In [27]:
test = [list(df.Men[:-1]), list(df.Women[:-1])]
test

[[2792, 1486], [3591, 2131]]

In [28]:
chi_calculated = scipy.stats.chi2_contingency(test)[0]
Dof = scipy.stats.chi2_contingency(test)[2]

# Suppose the level of significance is 5%.
chi_expected = scipy.stats.chi2.ppf(1-0.05, df = Dof)

print(f"The calculated chi square value is {chi_calculated}.")
print(f"The expected chi square value should be less than {chi_expected} in order to accept the null hypothesis.")

The calculated chi square value is 6.552337973968181.
The expected chi square value should be less than 3.841458820694124 in order to accept the null hypothesis.


###### Conclusion
Since calculated chi-calculated (i.e 6.55233) is more than the chi-critical (i.e 3.8414).  
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance in order to prove that 'sex' and 'voting' are dependent on each other.

# Problem 19:

A sample of 100 voters are asked which of four candidates they would vote for in an
election. The number supporting each candidate is given below:  
Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96,
with 3 df, 𝑝 = 0.05].

In [29]:
data = {
    'Candidate' : ['Higgins', 'Reardon', 'White', 'Chartton'],
    'Vote_freq(O)' : [41, 19, 24, 16]
}
df = pd.DataFrame(data)
df

Candidate  Vote_freq(O)
0   Higgins            41
1   Reardon            19
2     White            24
3  Chartton            16

##### Hypothesis
H0 : All the candidates are equally popular.  
H1 : All the candidates are not equally popular.

In [30]:
# Since the number of votera are equally popular in a given population.
df['Expected_votes(E)'] = 100//4
df

Candidate  Vote_freq(O)  Expected_votes(E)
0   Higgins            41                 25
1   Reardon            19                 25
2     White            24                 25
3  Chartton            16                 25

In [31]:
df['(E-O)^2'] = (df['Vote_freq(O)'] - df['Expected_votes(E)'])**2
df

Candidate  Vote_freq(O)  Expected_votes(E)  (E-O)^2
0   Higgins            41                 25      256
1   Reardon            19                 25       36
2     White            24                 25        1
3  Chartton            16                 25       81

In [32]:
df['(E-O)^2/E'] = df['(E-O)^2']/df['Expected_votes(E)']
df

Candidate  Vote_freq(O)  Expected_votes(E)  (E-O)^2  (E-O)^2/E
0   Higgins            41                 25      256      10.24
1   Reardon            19                 25       36       1.44
2     White            24                 25        1       0.04
3  Chartton            16                 25       81       3.24

In [33]:
chi_calculated = df['(E-O)^2/E'].sum()

# Since number of candidates are 4 hence 
Dof = 4-1

# Considering 5% level of significance 
p = 0.05
chi_critical = scipy.stats.chi2.ppf(1-p, df=Dof)

print(f"The chi-square calculated value is {chi_calculated}")
print(f"The chi-square should have value less than {chi_expected} in order to acccept the null hypothesis.")

The chi-square calculated value is 14.959999999999999
The chi-square should have value less than 3.841458820694124 in order to acccept the null hypothesis.


##### Conclusion
Since calculated chi-calculated (i.e 14.9599) is more than the chi-critical (i.e 3.8414).   
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance in order to prove that all the four candidates are NOT equally popular.

# Problem 20:

Children of three ages are asked to indicate their preference for three photographs of
adults. Do the data suggest that there is a significant relationship between age and
photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4
df: 𝑝 < 0.05].

In [34]:
data = {
    'Age of child' : ['5-6 years', '7-8 years', '9-10 years'],
    'A' : [18, 2 , 20],
    'B' : [22, 28, 10],
    'C' : [20, 40, 40]
}
df = pd.DataFrame(data)
df

Age of child   A   B   C
0    5-6 years  18  22  20
1    7-8 years   2  28  40
2   9-10 years  20  10  40

##### Hypothesis
H0 : Age and photograph are dependent of eachother.  
H1 : Age and photgraph are independent of eachother.

In [35]:
df.loc[len(df.index)] = ['Totals', df['A'].sum(), df['B'].sum(), df['C'].sum()]
df['Totals'] = df['A'] + df['B'] + df['C']
df

Age of child   A   B    C  Totals
0    5-6 years  18  22   20      60
1    7-8 years   2  28   40      70
2   9-10 years  20  10   40      70
3       Totals  40  60  100     200

In [36]:
test_data = [list(df.A[:-1]), list(df.B[:-1]), list(df.C[:-1])]
print(test_data)

[[18, 2, 20], [22, 28, 10], [20, 40, 40]]


In [37]:
chi_calculated = scipy.stats.chi2_contingency(test_data)[0]

Dof = scipy.stats.chi2_contingency(test_data)[2]

# Considering 5% level of significance
p = 0.05
chi_critical = scipy.stats.chi2.ppf(1-p, df = Dof)

print(f"The calculated value of chi-square is {chi_calculated}")
print(f"The chi-square should have value less than {chi_critical} in order to accept null hypothesis.")

The calculated value of chi-square is 29.603174603174605
The chi-square should have value less than 9.487729036781154 in order to accept null hypothesis.


##### Conclusion
Since calculated chi-calculated (i.e 29.6031) is more than the chi-critical (i.e 9.4877).    
Hence the null hypothesis is rejected.    
There is enough evidence at 5% level of significance in order to prove that 'age of child' and 'photograph selection' are independent.

# Problem 21:

A study of conformity using the Asch paradigm involved two conditions: one where
one confederate supported the true judgement and another where no confederate
gave the correct response.  
Is there a significant difference between the "support" and "no support" conditions in the
frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df:
𝑝 < 0.05].

In [38]:
data = {
    'Confrom status' : ['Conform', 'Not conform'],
    'Support' : [18, 32],
    'No Support' : [40, 10]
}

df = pd.DataFrame(data)
df

Confrom status  Support  No Support
0        Conform       18          40
1    Not conform       32          10

##### Hypothesis
H0 : Significant difference between the "support" and "no support".  
H1 : No significant difference between "support" and "no support".

In [39]:
df.loc[len(df.index)] = ['Total', df['Support'].sum(), df['No Support'].sum()]
df['Total'] = df['Support'] + df['No Support']
df

Confrom status  Support  No Support  Total
0        Conform       18          40     58
1    Not conform       32          10     42
2          Total       50          50    100

In [40]:
test_data = [list(df['Support'][:-1]), list(df['No Support'][:-1])]
print(test_data)

[[18, 32], [40, 10]]


In [41]:
chi_calculated = scipy.stats.chi2_contingency(test_data)[0]
Dof = scipy.stats.chi2_contingency(test_data)[2]

# Considering 5% level of significance
p = 0.05
chi_critical = scipy.stats.chi2.ppf(1-p, df=Dof)

print(f"The calculated value of chi-square is {chi_calculated}")
print(f"The chi-squared value should be less than {chi_critical} in order to accept the null hypotheis.")

The calculated value of chi-square is 18.10344827586207
The chi-squared value should be less than 3.841458820694124 in order to accept the null hypotheis.


##### Conclusion
Since calculated chi-calculated (i.e 18.1034) is more than the chi-critical (i.e 3.8414).    
Hence the null hypothesis is rejected.    
There is enough evidence at 5% level of significance in order to prove that there is no significant difference between "support" and "no support" conditions in the frequency with which individuals are likely to conform.

# Problem 22:

We want to test whether short people differ with respect to their leadership qualities
(Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget
MP's are there?) The following table shows the frequencies with which 43 short people and
52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a
relationship between height and leadership qualities?
[Chi-Square = 10.71, with 2 df: 𝑝 < 0.01].

In [42]:
data = {
    'Category' : ['Leader', 'Follower', 'Unclassifiable'],
    'Short' : [12, 22, 9],
    'Tall' : [32, 14, 6]
}

df = pd.DataFrame(data)
df

Category  Short  Tall
0          Leader     12    32
1        Follower     22    14
2  Unclassifiable      9     6

##### Hypothesis
H0 : There is no relationship between 'Height' and 'Leadership qualities'.    
H0 : There is relationship between 'Height' and 'Leadership qualities'.

In [43]:
df.loc[len(df.index)] = ['Total', df['Short'].sum(), df['Tall'].sum()]
df['Total'] = df['Short'] + df['Tall']
df

Category  Short  Tall  Total
0          Leader     12    32     44
1        Follower     22    14     36
2  Unclassifiable      9     6     15
3           Total     43    52     95

In [44]:
test_data = [list(df['Short'][:-1]), list(df['Tall'][:-1])]
test_data

[[12, 22, 9], [32, 14, 6]]

In [45]:
chi_calculated = scipy.stats.chi2_contingency(test_data)[0]

Dof = scipy.stats.chi2_contingency(test_data)[2]

# Considering 5% level of significance
p = 0.05
chi_critical = scipy.stats.chi2.ppf(1-p, df=Dof)

print(f"The calculated value of chi-square is {chi_calculated}.")
print(f"The chi-squared vale should be less than {chi_critical} in order to accept null hypothesis.")

The calculated value of chi-square is 10.712198008709638.
The chi-squared vale should be less than 5.991464547107979 in order to accept null hypothesis.


##### Conclusion
Since calculated chi-calculated (i.e 10.7121) is more than the chi-critical (i.e 5.9914).    
Hence the null hypothesis is rejected.    
There is enough evidence at 5% level of significance in order to prove that there is relationship between 'Height' and 'Leadership qualities'.

# Problem 23:

Each respondent in the Current Population Survey of March 1993 was classified as
employed, unemployed, or outside the labor force. The results for men in California age 35-
44 can be cross-tabulated by marital status, as follows:  
Men of different marital status seem to have different distributions of labor force status. Or is
this just chance variation? (you may assume the table results from a simple random
sample.)

In [46]:
data = {
    'Employment_status' : ['Employed', 'Unemployed', 'Not in labour force'],
    'Married' : [679, 63, 42],
    'Widowed, divorced or separated' : [103, 10, 18],
    'Never married' : [114, 20, 25]
}
df = pd.DataFrame(data)
df

Employment_status  Married  Widowed, divorced or separated  Never married
0             Employed      679                             103            114
1           Unemployed       63                              10             20
2  Not in labour force       42                              18             25

##### Hypothesis
H0 : Men of different marital status and different distributions of labour force are independent.  
H1 : Men of different marital status and different distributions of labour force are dependent.

In [47]:
df.loc[len(df.index)] = ['Total', df['Married'].sum(), df['Widowed, divorced or separated'].sum(), df['Never married'].sum()]
df['Total'] = df['Married'] + df['Widowed, divorced or separated'] + df['Never married']
df

Employment_status  Married  Widowed, divorced or separated  \
0             Employed      679                             103   
1           Unemployed       63                              10   
2  Not in labour force       42                              18   
3                Total      784                             131   

   Never married  Total  
0            114    896  
1             20     93  
2             25     85  
3            159   1074

In [48]:
data = [list(df['Married'][:-1]), list(df['Widowed, divorced or separated'][:-1]), list(df['Never married'][:-1])]
data

[[679, 63, 42], [103, 10, 18], [114, 20, 25]]

In [49]:
chi_calculated = scipy.stats.chi2_contingency(data)[0]
Dof = scipy.stats.chi2_contingency(data)[2]

# Considering 5% level of significance
p = 0.05
chi_critical = scipy.stats.chi2.ppf(1-p, df=Dof)

print(f"The calculated value of chi-square is {chi_calculated}.")
print(f"The chi-squared value should be less than {chi_critical} in order to accept null hypothesis.")

The calculated value of chi-square is 31.61310319407798.
The chi-squared value should be less than 9.487729036781154 in order to accept null hypothesis.


##### Conclusion
Since calculated chi-calculated (i.e 31.6131) is more than the chi-critical (i.e 9.4877).  
Hence the null hypothesis is rejected.  
There is enough evidence at 5% level of significance in order to prove that men of different marital status and different distribution of labour force are dependent.